### PR TITLE
CY-1905 Enable auto configuration for the manager status reporter

### PR DESCRIPTION
**Pull request by davidginzbourg**
- Some general refactoring

### CODE REVIEW

File: cfy_manager/components/restservice/restservice.py [hunk 1]
```diff
@@ -26,10 +26,21 @@
...
+from ...constants import (
+    REST_HOME_DIR,
+    REST_CONFIG_PATH,
+    SELECT_USER_TOKENS_QUERY,
+    REST_SECURITY_CONFIG_PATH,
+    REST_AUTHORIZATION_CONFIG_PATH
+)
 from ..components_constants import (
     VENV,
+    TOKEN,
```
Comment (tyacbovi) [inline cfy_manager/components/restservice/restservice.py]: there is a const for this

File: cfy_manager/components/restservice/restservice.py [hunk 2]
```diff
@@ -262,6 +269,22 @@ def _configure_db(self):
                 db.create_amqp_resources(configs)
             else:
                 self._join_cluster(configs)
+            self._fetch_manager_reporter_token()
+
+    @staticmethod
+    def _fetch_manager_reporter_token():
+        sql_stmnt = SELECT_USER_TOKENS_QUERY + " = '{0}'".format(
```
Comment (tyacbovi) [inline cfy_manager/components/restservice/restservice.py]: you can lose the '+' and just do the format

File: .circleci/config.yml [hunk 3]
```diff
@@ -105,6 +105,10 @@ jobs:
             cd packaging/docker
             chmod +x install_manager.sh
             ./install_manager.sh
+      - run:
```
Comment (tyacbovi) [inline .circleci/config.yml]: Not used?

File: .circleci/config.yml [hunk 4]
```diff
@@ -224,9 +228,7 @@ jobs:
       - run:
           name: status
```
Comment (tyacbovi) [inline .circleci/config.yml]: Update name

File: cfy_manager/components/restservice/restservice.py [hunk 5]
```diff
@@ -482,9 +505,20 @@ def configure(self):
...
 
         logger.notice('Rest Service successfully configured')
 
+    @staticmethod
+    def _configure_status_reporter():
+        conf = {
+            'token': config[MANAGER_STATUS_REPORTER][TOKEN],
+            'user_name': MANAGER_STATUS_REPORTER,
+            'managers_ips': ['localhost'],
+            'ca_path': config[SSL_INPUTS]['ca_cert_path']
```
[the remark this answers is not in the document]
Comment (davidginzbourg) [reply cfy_manager/components/restservice/restservice.py]: Changed

File: cfy_manager/components/restservice/restservice.py (hunk 1 above)
Comment (davidginzbourg) [reply cfy_manager/components/restservice/restservice.py]: Changed

File: cfy_manager/components/restservice/restservice.py (hunk 2 above)
Comment (davidginzbourg) [reply cfy_manager/components/restservice/restservice.py]: Changed

File: .circleci/config.yml (hunk 3 above)
Comment (davidginzbourg) [reply .circleci/config.yml]: Changed

File: .circleci/config.yml (hunk 4 above)
Comment (davidginzbourg) [reply .circleci/config.yml]: Changed

File: .circleci/cluster/validate_status.sh [hunk 6]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+set -eux
+
+validate_status() {
+  manager_status="$(docker exec $1 cfy cluster status)"
+  set +e
+  grep --ignore-case -q -e 'fail' -e 'error' <<< "$manager_status" && (echo "Found a failure in the cluster status" && exit 1)
```
Comment (tyacbovi) [inline .circleci/cluster/validate_status.sh]: you can change it to a little shorter command 'grep -iq'

File: cfy_manager/components/restservice/db.py [hunk 7]
```diff
@@ -222,32 +222,22 @@ def _create_process_env(rest_config=None, authorization_config=None,
...
-    :param args_dict: script arguments to pass to the script.
+    :param script_input: script input to pass to the script.
     :param configs: keword arguments dict to pass to _create_process_env(..).
     :return: the script's returned when it finished its execution.
     """
-    env_dict = None
-    if configs is not None:
-        env_dict = _create_process_env(**configs)
+    configs = configs or {}
+    env_dict = _create_process_env(**configs)
```
Comment (tyacbovi) [inline cfy_manager/components/restservice/db.py]: it could not stay None?

File: cfy_manager/main.py [hunk 8]
```diff
@@ -550,24 +505,24 @@ def _print_finish_message():
 def print_credentials_to_screen():
     password = config[MANAGER][SECURITY][ADMIN_PASSWORD]
     manager_status_reporter_token = config.get(
-        MANAGER_STATUS_REPORTER, {}).get(TOKEN, None)
+        MANAGER_STATUS_REPORTER, {}).get(STATUS_REPORTER_TOKEN, None)
```
Comment (tyacbovi) [inline cfy_manager/main.py]: You can remove the None-s, that's the default

File: cfy_manager/status_reporter/status_reporter.py [hunk 9]
```diff
@@ -17,40 +17,37 @@
 
 import json
 import logging
-from os import environ
-from os.path import join
 
 import argh
 
+from cfy_manager.utils.common import output_table
```
Comment (tyacbovi) [inline cfy_manager/status_reporter/status_reporter.py]: order

File: cfy_manager/status_reporter/status_reporter.py [hunk 10]
```diff
@@ -17,40 +17,37 @@
...
     MANAGER_STATUS_REPORTER
 )
-VERBOSE_HELP_MSG = "Used to give more verbose output."
-SCRIPTS_PATH = join(STATUS_REPORTER_DIR, 'scripts')
 
 logger = get_logger(STATUS_REPORTER)
 
 
-@argh.arg('--managers-ip',
+@argh.arg('--managers-ips',
```
Comment (tyacbovi) [inline cfy_manager/status_reporter/status_reporter.py]: Yoni and Mohammed need to know about this change

File: cfy_manager/status_reporter/status_reporter.py [hunk 11]
```diff
@@ -185,40 +181,36 @@ def get_tokens(json_format=None, verbose=False):
...
+    setup_console_logger(verbose=verbose)
+    logger.info('Fetching config...')
+    _config = read_yaml_file(STATUS_REPORTER_CONFIGURATION_PATH)
+    if not _config:
+        logger.error("Config file doesn't exist.")
+        return
+    if json_format:
+        print(json.dumps(_config))
+    else:
+        logger.info('Config loaded')
```
Comment (tyacbovi) [inline cfy_manager/status_reporter/status_reporter.py]: not really needed

File: cfy_manager/utils/scripts.py [hunk 12]
```diff
@@ -13,22 +13,28 @@
...
 from ..utils import common
 from ..logger import get_logger
-from ..constants import REST_HOME_DIR, SCRIPTS
 from ..utils.files import write_to_tempfile
+from cfy_manager.exceptions import FileError
+from ..constants import REST_HOME_DIR, SCRIPTS, REST_SECURITY_CONFIG_PATH
 
 logger = get_logger(SCRIPTS)
 
+THIS_DIR = dirname(__file__)
```
Comment (tyacbovi) [inline cfy_manager/utils/scripts.py]: CURRENT_DIR

File: .circleci/cluster/validate_status.sh (hunk 6 above)
Comment (davidginzbourg) [reply .circleci/cluster/validate_status.sh]: 200 iq comment

File: cfy_manager/components/restservice/db.py (hunk 7 above)
Comment (davidginzbourg) [reply cfy_manager/components/restservice/db.py]: Changed to `_create_process_env(**configs) if configs else None`

File: cfy_manager/main.py (hunk 8 above)
Comment (davidginzbourg) [reply cfy_manager/main.py]: You're right, removed.

File: cfy_manager/status_reporter/status_reporter.py (hunk 9 above)
Comment (davidginzbourg) [reply cfy_manager/status_reporter/status_reporter.py]: Reordered

File: cfy_manager/status_reporter/status_reporter.py (hunk 10 above)
Comment (davidginzbourg) [reply cfy_manager/status_reporter/status_reporter.py]: I'll let them know personally

Comment (davidginzbourg) [reply cfy_manager/status_reporter/status_reporter.py]: Cancelled. I'll create a JIRA for it

**Pull request merged**